helloworld/main.py

```diff
@@ -1,4 +1,62 @@
-from flask import escape
+import os
+import io
+from flask import escape, request
+from flask_restful import Resource, Api
+from google.cloud import vision
+
+from flask import Flask, jsonify
+from flask_cors import CORS
+
+os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = r'GVServiceAccountToken.json'
+
+app = Flask(__name__)
+api = Api(app)
+
+# Set this to false when deploying to live application
+app.config['DEBUG'] = True
+
+# Allow cross-origin resource sharing
+cors = CORS(app, resources={r'/api/*': {'origins': '*'}})
+
+
+def face_detection(uri):
+    """HTTP Cloud Function.
+    Args:
+        request (flask.Request): The request object.
+        <https://flask.palletsprojects.com/en/1.1.x/api/#incoming-request-data>
+    Returns:
+        The response text, or any set of values that can be turned into a
+        Response object using `make_response`
+        <https://flask.palletsprojects.com/en/1.1.x/api/#flask.make_response>.
+    """
+    # _json = request.json
+    # _uri = _json['uri']
+    # print(_uri)
+    vision_client = vision.ImageAnnotatorClient()
+    print(vision_client)
+
+    # image = vision.types.Image()
+    image = vision.Image()
+    image.source.image_uri = uri
+
+    response = vision_client.face_detection(image=image)
+    faceAnnotations = response.face_annotations
+
+    likelihood = ('unknown', 'Very unlikely', 'Unlikely',
```

Comment:
could you please add comments on what this code is doing?

Reply:
I will put in comments and upload it again. But in this function I am just trying to ensure that Google vision is successfully acting upon when it is provided an image via url. Currently likelihood list is there provide labels when printing an output.
Output example:
Detection Confidence:  0.8303185701370239
Angry likelyhood:  Very unlikely
Joy likelyhood:  Very likely
Sorrow likelyhood:  Very unlikely
Surprise likelyhood:  Very unlikely
Headwear likelyhood:  Very unlikely